nshutijonathan/Auto-Mart
# #166348359 Build out Security measures in Auto-Mart project

#### What does this PR do?
Have JSON web tokens generated  
Have protected the endpoints with JSON web tokens

#### Description of Task to be completed?
Implemented role-based authorization

Have The following endpoint working and protected with web tokens
* GET/localhost:3000/api/v1/users
*   GET/localhost:3000/api/v1/users/<:id>
*   GET/ localhost:3000/api/v1/orders
*   DELETE/localhost:3000/api/v1/users/<:id>
*   POST/localhost:3000/api/v1/car
*   PUT/localhost:3000/api/v1/order/<:id>/price
*   PUT/localhost:3000/api/v1/car/<:id>/status
*   PUT/localhost:3000/api/v1/car/<:id>/price
*   DELETE/ localhost:3000/api/v1/cars/<:id>

#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with data structures to persist some data
#### What are the relevant pivotal tracker stories?
[#166348359](https://www.pivotaltracker.com/n/projects/2346053/stories/166348359)

#### Screenshots
![auth](https://user-images.githubusercontent.com/16412410/58688015-1f6aa200-8338-11e9-9f78-50d8e27056c9.PNG)
![deleted](https://user-images.githubusercontent.com/16412410/58688019-24c7ec80-8338-11e9-8101-5be7dd43d61a.PNG)

--
